pyanalyze/options.py

```diff
@@ -371,6 +375,11 @@ def parse_config_file(
     )
 
 
+@functools.lru_cache
+def get_all_error_codes() -> FrozenSet[str]:
```

Comment:
I do not really like this function. I think that, ideally, this should be done by using the `registry` option populated within `__init_subclass__` created by `ConfigOption`. However, all the error codes are deriving from `BooleanOptions`. But not only error codes, but also some other boolean based options.

A solution could be creating a new class deriving from `BooleanOptions` that would only contain all the error codes in `registry`. This way, the logic splits the boolean options between error-based and pure config based. And therefore, `registry` would be populated with our error codes. I am not sure if this makes sense, but it is also a bigger rework that might not be interesting. Maybe you know better ways of accessing all the error codes without using this function that I wrote :)

Reply:
I think that's actually a good idea and shouldn't be too difficult to implement: add a new subclass of BooleanOption that is only for error codes. Might also play better with pyanalyze's functionality for registering new error codes from plugins.

Reply:
Thinking about it more that feels like unnecessary extra complexity. I'll push a few changes and then merge this.